scipy/stats/tests/test_stats.py

```diff
@@ -7124,10 +7126,16 @@ def test_raises_value_error_non_numeric_input(self):
         with pytest.raises(TypeError, match="ufunc 'log' not supported"):
             stats.gstd('You cannot take the logarithm of a string.')
 
-    @pytest.mark.parametrize('bad_value', (0, -1, np.inf))
+    @pytest.mark.parametrize('bad_value', (0, -1, np.inf, np.nan))
```

Comment:
In 1.13, non-positive and infinte input produced different errors, and `np.nan` produced no warning.

After gh-20696, none of these produced warnings or errors. So we get to pick which emit warnings and which don't.

We discussed separately that negative values should produce a warning.
I don't think `np.nan` should produce a warning because we follow NumPy's example, propagating NaNs silently in all stats functions.
`np.inf` is an unusual input, so it is not so surprising that the (correct) result would be NaN, but I could be convinced that a warning is needed nonetheless. I know that `np.std` and `stats.moment` produce a warning in this case, so maybe this should, too.

Reply:
I'm indifferent about a warning for `np.inf`. Sounds reasonable otherwise.

Reply:
We can always come back if there is a need. For instance, perhaps it will be requested in my `lmoment` PR, in which case I'll take care of it here. I suppose we'll leave it for now unless there is a strong opinion.